client/asset/btc/electrum_client.go

```diff
@@ -0,0 +1,1289 @@
+	txid := txHash.String()
+	// Try the wallet first in case this is a wallet transaction (own swap).
+	txRaw, confs, err := ew.checkWalletTx(txid)
+	if err == nil {
+		msgTx, err := msgTxFromBytes(txRaw)
+		if err != nil {
+			return 0, false, err
+		}
+		if vout >= uint32(len(msgTx.TxOut)) {
+			return 0, false, fmt.Errorf("output %d of tx %v does not exists", vout, txid)
+		}
+		pkScript = msgTx.TxOut[vout].PkScript
+	} else {
+		// Fall back to the more expensive server request.
+		txInfo, err := ew.chain.GetTransaction(context.Background(), txid)
+		if err != nil {
+			return 0, false, err
+		}
+		confs = uint32(txInfo.Confirmations)
+		if txInfo.Confirmations < 1 {
+			confs = 0
+		}
+		if vout >= uint32(len(txInfo.Vout)) {
+			return 0, false, fmt.Errorf("output %d of tx %v does not exists", vout, txid)
+		}
+		txOut := &txInfo.Vout[vout]
+		pkScript, err = hex.DecodeString(txOut.PkScript.Hex)
+		if err != nil {
+			return 0, false, fmt.Errorf("invalid pkScript: %w", err)
+		}
+	}
+
+	// Decode the pkScript and extract the address.
+	scriptClass, addrs, _, err := txscript.ExtractPkScriptAddrs(pkScript, ew.chainParams)
+	if err != nil {
+		return 0, false, fmt.Errorf("invalid pkScript: %v", err)
+	}
+	switch scriptClass { // require p2sh or p2wsh... necessary?
+	case txscript.ScriptHashTy, txscript.WitnessV0ScriptHashTy:
+	default:
+		return 0, false, fmt.Errorf("pkScript not a script hash: %v", scriptClass)
+	}
+	if len(addrs) != 1 {
+		return 0, false, fmt.Errorf("invalid pkScript: %d addresses", len(addrs))
+	}
+	swapAddr, err := ew.stringAddr(addrs[0], ew.chainParams)
+	if len(addrs) != 1 {
+		return 0, false, fmt.Errorf("invalid swap address encoding: %w", err)
+	}
+
+	// Now see if the unspent outputs for this address include this outpoint.
+	addrUnspents, err := ew.wallet.GetAddressUnspent(swapAddr)
+	if err != nil {
+		return 0, false, fmt.Errorf("getaddressunspent: %v", err)
+	}
+	spent = true
+	for _, utxo := range addrUnspents {
+		if utxo.TxHash == txid && uint32(utxo.TxPos) == vout {
+			spent = false
+			break
+		}
+	}
+	return confs, spent, nil
+}
+
+func (ew *electrumWallet) sendWithSubtract(address string, value, feeRate uint64) (*chainhash.Hash, error) {
+	addr, err := ew.decodeAddr(address, ew.chainParams)
+	if err != nil {
+		return nil, err
+	}
+	pkScript, err := txscript.PayToAddrScript(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	unfundedTxSize := dexbtc.MinimumTxOverhead + dexbtc.P2WPKHOutputSize /* change */ +
+		dexbtc.TxOutOverhead + uint64(len(pkScript)) // send-to address
+
+	unspents, err := ew.listUnspent()
+	if err != nil {
+		return nil, fmt.Errorf("error listing unspent outputs: %w", err)
+	}
+
+	// Filter out locked outpoints since listUnspent includes them.
+	unspentMap := make(map[RPCOutpoint]*ListUnspentResult, len(unspents))
+	for _, unspent := range unspents {
+		unspentMap[RPCOutpoint{unspent.TxID, unspent.Vout}] = unspent
+	}
+	locked, _ := ew.listLockUnspent() // does not error
```

Comment:
The implementation might change later and this would still not be checking for an error.